Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -2175,9 +2188,11 @@ private void UpdateResourceAvailability(PerformanceCollector.MachinePerfInfo per
 
                 if (!m_scheduleConfiguration.DisableProcessRetryOnResourceExhaustion)
                 {
-                    // Free down to the specified max RAM utilization percentage with 10% slack
-                    var desiredRamToFreePercentage =
-                        (perfInfo.RamUsagePercentage.Value - m_configuration.Schedule.MaximumRamUtilizationPercentage) + 10;
+                    // Free down to the specified max RAM utilization percentage with 10% slack, if VM memory pressure is critical
+                    // we instantly try to free up 15% of the overall system memory to improve stability
+                    var desiredRamToFreePercentage = !isMemoryPressureCritical
+                        ? (perfInfo.RamUsagePercentage.Value - m_configuration.Schedule.MaximumRamUtilizationPercentage) + 10
+                        : 15;

```

Comment:
10% vs 15%: Do we really need different thresholds based on the OS? Can we just utilize the externally configurable MaximumRamUtilizationPercentage option instead? I feel like internally padding the number differently based on OS is a bit unexpected from end user perspective.